UnityAuth/src/main/java/io/unityfoundation/auth/AuthController.java

```diff
@@ -110,6 +104,43 @@ public HttpResponse<HasPermissionResponse> hasPermission(@Body HasPermissionRequ
     return createHasPermissionResponse(true, user.getEmail(), null, commonPermissions);
   }
 
+  @Get("/roles")
+  public HttpResponse<List<RoleDTO>> getRoles() {
+    return HttpResponse.ok(roleRepo.findAll().stream()
+            .map(role -> new RoleDTO(role.getId(), role.getName(), role.getDescription()))
+            .toList());
+  }
+
+  @Get("/tenants")
+  public HttpResponse<List<TenantDTO>> getTenants(Authentication authentication) {
+
+    String authenticatedUserEmail = authentication.getName();
+
+    List<Tenant> tenants = userRepo.existsByEmailAndRoleEqualsUnityAdmin(authenticatedUserEmail) ?
+            tenantRepo.findAll() : tenantRepo.findAllByUserEmail(authenticatedUserEmail);
+
+    return HttpResponse.ok(tenants.stream()
+            .map(tenant -> new TenantDTO(tenant.getId(), tenant.getName()))
+            .toList());
+  }
+
+  @Get("/tenants/{id}/users")
```

Comment:
Who is allowed to call this endpoint?

Reply:
Only logged in users. Should we limit based on role?